aries-site/src/examples/components/pagination/PaginationServerTableExample.js

```diff
@@ -0,0 +1,172 @@
+    render: datum => {
+      if (datum.success === false) {
+        const content = (
+          <Box width={{ max: 'medium' }}>
+            {datum.failures?.map(({ reason }) => (
+              <Text key={reason}>{reason}</Text>
+            ))}
+          </Box>
+        );
+        return (
+          <Tip
+            plain
+            content={content}
+            dropProps={{
+              round: 'medium',
+              pad: 'small',
+              background: 'background-back',
+            }}
+          >
+            <Box>
+              <StatusCritical color="red" />
+            </Box>
+          </Tip>
+        );
+      }
+      return undefined;
+    },
+  },
+];
+
+export const PaginationServerTableExample = () => {
+  const [sort, setSort] = useState({ property: 'name', direction: 'asc' });
+  const [data, setData] = useState([]);
+  const [numberItems, setNumberItems] = useState(0);
+  const [page, setPage] = useState(1);
+  const [limit, setLimit] = useState(10);
+  const pageResultStart = (page - 1) * limit + 1;
+  const pageResultEnd = Math.min(page * limit, numberItems);
+  const options = [5, 10, 25, 50, 100];
+
+  useEffect(() => {
+    const fetchData = async () => {
+      const query = {
+        options: {
+          populate: [
+            {
+              path: 'rocket',
+              select: { name: 1 },
+            },
+          ],
+          sort: {
+            [sort.property || 'name']: sort.direction || 'asc',
+          },
+          select: ['name', 'success', 'failures'],
+          limit,
+          page,
+        },
+      };
+      fetch('https://api.spacexdata.com/v4/launches/query', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(query),
+      })
+        .then(response => response.json())
+        .then(d => {
+          setPage(d.page);
+          setNumberItems(d.totalDocs);
+          setData(d.docs || []);
+        })
+        .catch(error => console.error('Unable to get data:', error));
+    };
+    fetchData();
+  }, [sort, page, limit]);
+
+  return (
+    <Box pad="small">
+      <Heading id="server-side-pagination-heading" level={3}>
+        Launches
+      </Heading>
+
+      {numberItems > limit && (
+        <Box
+          pad={{ vetical: 'small' }}
+          align="center"
+          direction="row-responsive"
+          gap="small"
+        >
+          <Text>Select</Text>
+          <Select
+            id="select"
+            name="select"
+            placeholder="10"
+            value={limit}
+            options={options}
+            onChange={({ option }) => setLimit(option)}
+          />
+          <Text>entries</Text>
+        </Box>
+      )}
+      <Box>
+        <DataTable
+          aria-describedby="server-side-pagination-heading"
+          columns={columns}
+          data={data}
+          sort={{ ...sort, external: true }}
+          replace
+          step={limit}
```

Comment:
Remove the `replace` and the `step`. You don't need either since you're not doing infinitescroll and you're handling the step and pagination yourself. I think this fixes the issue you saw with selecting a smaller step in the select and the table going empty. Not sure why though. Sounds like a potential DataTable bug

Reply:
It turns out that the issue was with the `replace` prop! It makes sense to remove the `step` prop as well, since it is being handled below in the Pagination component. Thanks for your feedback!